Sync `common` from WPT upstream

Ahmad-S792 opened the pull request:
#### fd04cf641b1a2409f967a4b354211b84c6bc7bc9
<pre>
Sync `common` from WPT upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=294074">https://bugs.webkit.org/show_bug.cgi?id=294074</a>
<a href="https://rdar.apple.com/152656968">rdar://152656968</a>

Reviewed by Anne van Kesteren.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/a8eadf139cc69e24c3d5b7c271be529a6d1e08ee">https://github.com/web-platform-tests/wpt/commit/a8eadf139cc69e24c3d5b7c271be529a6d1e08ee</a>

This patch does not sync `get-host-info.sub.js` because it will undo changes
done in 254273@main. It also does not sync `media.js`, which was leading
to lot of test failures.

* LayoutTests/imported/w3c/web-platform-tests/common/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/common/custom-cors-response.js:
(const.custom_cors_response):
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/README.md:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/dispatcher.js:
(findLocationFromAncestors):
(findLocation):
(async await):
(const.sendItem):
(async while):
(async const.processQueue):
(async const):
(const.send):
(remoteExecutorUrl):
(RemoteContext.prototype.async execute_script):
(Executor.prototype.async execute):
(Executor):
(async const.receive): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/object-association.js:
(runTests):
(window.testIsPerWindow.propertyName.async_test.t.iframe.onload.t.step_func): Deleted.
(window.testIsPerWindow.propertyName.async_test.t.iframe.onload.t.step_func_done): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/redirect.py:
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/sab.js:
(const.createBuffer):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/resources/common.sub.js:
(const.subresourceMap.string_appeared_here.additionalAttributes.requestViaDedicatedWorker.workerUrlThatImports):
(const.subresourceMap.string_appeared_here.additionalAttributes.requestViaSharedWorker.workerUrlThatImports):
(invokeRequest):
(const.subresourceMap.string_appeared_here.invoker.url.requestViaDedicatedWorker.workerUrlThatImports): Deleted.
(const.subresourceMap.string_appeared_here.invoker.url.requestViaSharedWorker.workerUrlThatImports): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/scope/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/subresource/static-import.py:
(get_csp_value):
(main.payload_generator):
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/subresource/video.py:
(generate_payload):
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/generate.py:
(generate_test_source_files):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/spec_validator.py:
(assert_non_empty_string):
(assert_atom_or_list_items_from):
(assert_valid_spec_json):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/util.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/__init__.py:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/remote-executor-worker.js:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/remote-executor.html:
* LayoutTests/imported/w3c/web-platform-tests/common/dummy.json:
* LayoutTests/imported/w3c/web-platform-tests/common/proxy-all.sub.pac:
* LayoutTests/imported/w3c/web-platform-tests/common/refresh.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/__init__.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/scope/__init__.py:

&gt; Rebased Test Expectations:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/registries/per-global-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/system-state-and-capabilities/the-navigator-object/per-global.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/WebIDL/ecmascript-binding/allow-resizable-expected.txt:

Canonical link: <a href="https://commits.webkit.org/295910@main">https://commits.webkit.org/295910@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db645fbffff7816f7e0c87604e4756ce5b52178d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106550 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26301 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16698 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111754 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57146 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108589 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26971 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34802 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80908 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109554 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21384 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96140 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61243 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20835 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14244 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56589 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90723 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14277 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114616 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33688 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24825 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89979 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34052 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92371 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89689 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22886 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34595 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12404 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29297 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33613 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33359 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36712 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34958 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
Looks good modulo bots.

Ahmad-S792 committed:
Sync `common` from WPT upstream

https://bugs.webkit.org/show_bug.cgi?id=294074
rdar://152656968

Reviewed by Anne van Kesteren.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/a8eadf139cc69e24c3d5b7c271be529a6d1e08ee

This patch does not sync `get-host-info.sub.js` because it will undo changes
done in 254273@main. It also does not sync `media.js`, which was leading
to lot of test failures.

* LayoutTests/imported/w3c/web-platform-tests/common/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/common/custom-cors-response.js:
(const.custom_cors_response):
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/README.md:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/dispatcher.js:
(findLocationFromAncestors):
(findLocation):
(async await):
(const.sendItem):
(async while):
(async const.processQueue):
(async const):
(const.send):
(remoteExecutorUrl):
(RemoteContext.prototype.async execute_script):
(Executor.prototype.async execute):
(Executor):
(async const.receive): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/object-association.js:
(runTests):
(window.testIsPerWindow.propertyName.async_test.t.iframe.onload.t.step_func): Deleted.
(window.testIsPerWindow.propertyName.async_test.t.iframe.onload.t.step_func_done): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/redirect.py:
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/sab.js:
(const.createBuffer):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/resources/common.sub.js:
(const.subresourceMap.string_appeared_here.additionalAttributes.requestViaDedicatedWorker.workerUrlThatImports):
(const.subresourceMap.string_appeared_here.additionalAttributes.requestViaSharedWorker.workerUrlThatImports):
(invokeRequest):
(const.subresourceMap.string_appeared_here.invoker.url.requestViaDedicatedWorker.workerUrlThatImports): Deleted.
(const.subresourceMap.string_appeared_here.invoker.url.requestViaSharedWorker.workerUrlThatImports): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/scope/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/subresource/static-import.py:
(get_csp_value):
(main.payload_generator):
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/subresource/video.py:
(generate_payload):
(main):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/generate.py:
(generate_test_source_files):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/spec_validator.py:
(assert_non_empty_string):
(assert_atom_or_list_items_from):
(assert_valid_spec_json):
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/tools/util.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/common/__init__.py:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/remote-executor-worker.js:
* LayoutTests/imported/w3c/web-platform-tests/common/dispatcher/remote-executor.html:
* LayoutTests/imported/w3c/web-platform-tests/common/dummy.json:
* LayoutTests/imported/w3c/web-platform-tests/common/proxy-all.sub.pac:
* LayoutTests/imported/w3c/web-platform-tests/common/refresh.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/__init__.py:
* LayoutTests/imported/w3c/web-platform-tests/common/security-features/scope/__init__.py:

> Rebased Test Expectations:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/registries/per-global-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/system-state-and-capabilities/the-navigator-object/per-global.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/WebIDL/ecmascript-binding/allow-resizable-expected.txt:

Canonical link: https://commits.webkit.org/295910@main
